src/libraries/System.Runtime.Numerics/src/System/Numerics/BigInteger.cs

```diff
@@ -2785,5 +2837,1166 @@ private void AssertValid()
                 Debug.Assert(_sign > int.MinValue);
             }
         }
+
+        //
+        // IAdditionOperators
+        //
+
+        /// <inheritdoc cref="IAdditionOperators{TSelf, TOther, TResult}.op_Addition(TSelf, TOther)" />
+        static BigInteger IAdditionOperators<BigInteger, BigInteger, BigInteger>.operator checked +(BigInteger left, BigInteger right) => left + right;
+
+        //
+        // IAdditiveIdentity
+        //
+
+        /// <inheritdoc cref="IAdditiveIdentity{TSelf, TResult}.AdditiveIdentity" />
+        static BigInteger IAdditiveIdentity<BigInteger, BigInteger>.AdditiveIdentity => Zero;
+
+        //
+        // IBinaryInteger
+        //
+
+        /// <inheritdoc cref="IBinaryInteger{TSelf}.DivRem(TSelf, TSelf)" />
+        public static (BigInteger Quotient, BigInteger Remainder) DivRem(BigInteger left, BigInteger right)
+        {
+            BigInteger quotient = DivRem(left, right, out BigInteger remainder);
+            return (quotient, remainder);
+        }
+
+        /// <inheritdoc cref="IBinaryInteger{TSelf}.LeadingZeroCount(TSelf)" />
+        public static BigInteger LeadingZeroCount(BigInteger value)
```

Comment:
If you change `BigInteger` to use _nint\[\]_ instead of _int\[\]_ in a future PR (per your comment at https://github.com/dotnet/runtime/pull/68964/files#r866933097), you'll need to normalize the result of this method across platforms. Otherwise the results could differ on x86 vs x64.

Reply:
Yes. However, I believe that's fine/acceptable.

Various types, including user defined ones, may be variable sized. As such, `LeadingZeroCount` is logically a value between `0` and `GetByteCount() * 8L` (put another way it is `(GetByteCount() * 8L) - GetShortestBitLength()`.

This is functionally similar to how `nint`, `nuint`, `CLong`, and `CULong` work as well and gives a well-defined contract even in the face of this. The value is then still usable within these limits/expectations.